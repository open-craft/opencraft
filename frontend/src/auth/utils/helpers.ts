import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
