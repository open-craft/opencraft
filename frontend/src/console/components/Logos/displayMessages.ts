const messages = {
  logo: {
    defaultMessage: 'Logo',
    description: ''
  },
  logoDescription: {
    defaultMessage:
      'Your branding to be displayed throughout your instance. ' +
      "If unset, OpenCraft's logo will be used.",
    description: ''
  },
  uploadLogo: {
    defaultMessage: 'Upload Logo',
    description: ''
  },
  favicon: {
    defaultMessage: 'Favicon',
    description: ''
  },
  faviconDescription: {
    defaultMessage:
      'This is the icon that will be displayed in the browser tab. ' +
      "If unset, OpenCraft's logo will be used.",
    description: ''
  },
  uploadFavicon: {
    defaultMessage: 'Upload Favicon',
    description: ''
  }
};

export default messages;
