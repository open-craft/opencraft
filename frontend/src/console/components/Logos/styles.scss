@import "~styles/theme";

.custom-logo-pages {
  h2 {
    margin-bottom: 0px !important;
  }

  p {
    height: 17px;
    font-family: $chivo-font;
    font-size: 14px;
    letter-spacing: 0.14px;
    color: $secondary-4;
  }

  .image-container {
    display: flex;
    justify-content: center;
    vertical-align: middle;

    img {
      max-width: 100%;
      max-height: 100%;
    }
  }
}
