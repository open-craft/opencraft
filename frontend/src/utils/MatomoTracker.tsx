import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
