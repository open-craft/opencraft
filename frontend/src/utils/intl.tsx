import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
