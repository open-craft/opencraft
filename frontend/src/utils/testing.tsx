import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
