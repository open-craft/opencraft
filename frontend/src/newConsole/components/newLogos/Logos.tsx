import * as React from 'react';
import './styles.scss';
import { ConsolePage, PreviewBox } from 'newConsole/components';
import { InstancesModel } from 'console/models';
import { connect } from 'react-redux';
import { RootState } from 'global/state';

interface State {}
interface ActionProps {
  clearErrorMessage: Function;
  updateImages: Function;
}
interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {}

export class LogosComponent extends React.PureComponent<Props, State> {
  public render() {
    return (
      <ConsolePage contentLoading={this.props.loading}>
        <PreviewBox />
      </ConsolePage>
    );
  }
}

export const Logos = connect<StateProps, ActionProps, {}, Props, RootState>(
  (state: RootState) => state.console
)(LogosComponent);
