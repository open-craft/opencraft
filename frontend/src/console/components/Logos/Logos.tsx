import * as React from 'react';
import './styles.scss';
import {
  ConsolePage,
  ConsolePageCustomizationContainer
} from 'console/components';
import { Row, Col } from 'react-bootstrap';
import { InstancesModel } from 'console/models';
import { CollapseEditArea, ImageUploadField } from 'ui/components';
import { connect } from 'react-redux';
import { RootState } from 'global/state';
import { WrappedMessage } from 'utils/intl';
import { clearErrorMessage, updateImages } from 'console/actions';
import messages from './displayMessages';

interface State {}
interface ActionProps {
  clearErrorMessage: Function;
  updateImages: Function;
}
interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {}

export class LogosComponent extends React.PureComponent<Props, State> {
  updateImage = (imageName: string, image: File) => {
    if (this.props.activeInstance && this.props.activeInstance.data) {
      this.props.updateImages(
        this.props.activeInstance.data.id,
        imageName,
        image
      );
    }
  };

  public render() {
    const instance = this.props.activeInstance;

    return (
      <ConsolePage contentLoading={this.props.loading}>
        <div className="custom-logo-pages">
          <ConsolePageCustomizationContainer>
            <Row>
              <Col md={9}>
                <h2>
                  <WrappedMessage messages={messages} id="logo" />
                </h2>
                <p>
                  <WrappedMessage messages={messages} id="logoDescription" />
                </p>
              </Col>
              <Col md={3} className="image-container">
                <div>
                  {instance.data && instance.data.logo && (
                    <img src={instance.data.logo} alt="Logo" />
                  )}
                </div>
              </Col>
            </Row>
            <CollapseEditArea initialExpanded>
              <ImageUploadField
                customUploadMessage={
                  <WrappedMessage messages={messages} id="uploadLogo" />
                }
                updateImage={(image: File) => {
                  this.updateImage('logo', image);
                }}
                recommendedSize="48x48 px"
                error={instance.feedback.logo}
                clearError={() => {
                  this.props.clearErrorMessage('logo');
                }}
              />
            </CollapseEditArea>
          </ConsolePageCustomizationContainer>

          <ConsolePageCustomizationContainer>
            <Row>
              <Col md={9}>
                <h2>
                  <WrappedMessage messages={messages} id="favicon" />
                </h2>
                <p>
                  <WrappedMessage messages={messages} id="faviconDescription" />
                </p>
              </Col>
              <Col md={3} className="image-container">
                <div>
                  {instance.data && instance.data.favicon && (
                    <img src={instance.data.favicon} alt="favicon" />
                  )}
                </div>
              </Col>
            </Row>
            <CollapseEditArea>
              <ImageUploadField
                customUploadMessage={
                  <WrappedMessage messages={messages} id="uploadFavicon" />
                }
                updateImage={(image: File) => {
                  this.updateImage('favicon', image);
                }}
                error={instance.feedback.favicon}
                clearError={() => {
                  this.props.clearErrorMessage('favicon');
                }}
              />
            </CollapseEditArea>
          </ConsolePageCustomizationContainer>
        </div>
      </ConsolePage>
    );
  }
}

export const Logos = connect<StateProps, ActionProps, {}, Props, RootState>(
  (state: RootState) => state.console,
  {
    clearErrorMessage,
    updateImages
  }
)(LogosComponent);
