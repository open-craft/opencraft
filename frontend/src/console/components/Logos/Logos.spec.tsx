import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { Logos } from './Logos';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<Logos />).toJSON();
    expect(tree).toMatchSnapshot();
});
