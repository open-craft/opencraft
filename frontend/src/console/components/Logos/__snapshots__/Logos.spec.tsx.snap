// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="custom-logo-pages"
>
  <div
    className="console-page"
  >
    <div
      className="title-container"
    >
      <h1>
        <i
          className="fas fa-sync-alt fa-spin"
        />
      </h1>
    </div>
    <div
      className="d-flex justify-content-center align-middle redeployment-toolbar"
    >
      <div
        className="redeployment-nav"
      >
        <span
          className="status-pill badge badge-pill badge-primary"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="dot"
            style={
              Object {
                "backgroundColor": "grey",
              }
            }
          />
          <div
            className="text"
          >
            Status: loading
          </div>
        </span>
        <button
          className="float-right loading btn btn-primary btn-lg"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Deploy (0 updates)
        </button>
      </div>
    </div>
    <div
      className="console-page-container"
    >
      <div
        className="console-page-content row"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col-md-3"
            >
              <div
                className="customization-menu accordion"
              >
                <div
                  className="card"
                >
                  <div
                    className="card-header"
                  >
                    <div
                      className="card-header"
                      onClick={[Function]}
                      variant="link"
                    >
                      Theme
                    </div>
                  </div>
                  <div
                    aria-expanded={null}
                    className="collapse show"
                  >
                    <div
                      className="card-body"
                    >
                      <div
                        className="flex-column nav card-header-undefined"
                        onKeyDown={[Function]}
                      >
                        <a
                          aria-current={null}
                          href="/console/theming/preview-and-colors"
                          onClick={[Function]}
                        >
                          Preview & colors
                        </a>
                        <a
                          aria-current={null}
                          href="/console/theming/logos"
                          onClick={[Function]}
                        >
                          Logos
                        </a>
                        <a
                          aria-current={null}
                          href="/console/theming/buttons"
                          onClick={[Function]}
                        >
                          Buttons
                        </a>
                        <a
                          aria-current={null}
                          href="/console/theming/navigation"
                          onClick={[Function]}
                        >
                          Navigation
                        </a>
                        <a
                          aria-current={null}
                          href="/console/theming/footer"
                          onClick={[Function]}
                        >
                          Footer
                        </a>
                        <a
                          aria-current={null}
                          href="/console/theming/hero"
                          onClick={[Function]}
                        >
                          Hero
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="card"
                >
                  <div
                    className="card-header"
                  >
                    <div
                      className="card-header"
                      onClick={[Function]}
                      variant="link"
                    >
                      Custom Pages
                    </div>
                  </div>
                  <div
                    aria-expanded={null}
                    className="collapse"
                  >
                    <div
                      className="card-body"
                    >
                      <div
                        className="flex-column nav card-header-undefined"
                        onKeyDown={[Function]}
                      >
                        <a
                          aria-current={null}
                          href="/console/custom-pages/about"
                          onClick={[Function]}
                        >
                          About
                        </a>
                        <a
                          aria-current={null}
                          href="/console/custom-pages/contact"
                          onClick={[Function]}
                        >
                          Contact
                        </a>
                        <a
                          aria-current={null}
                          href="/console/custom-pages/donate"
                          onClick={[Function]}
                        >
                          Donate
                        </a>
                        <a
                          aria-current={null}
                          href="/console/custom-pages/tos"
                          onClick={[Function]}
                        >
                          Terms of Service
                        </a>
                        <a
                          aria-current={null}
                          href="/console/custom-pages/honor"
                          onClick={[Function]}
                        >
                          Honor Code
                        </a>
                        <a
                          aria-current={null}
                          href="/console/custom-pages/privacy"
                          onClick={[Function]}
                        >
                          Privacy Policy
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="card"
                >
                  <div
                    className="card-header"
                  >
                    <div
                      className="card-header"
                      onClick={[Function]}
                      variant="link"
                    >
                      Instance Settings
                    </div>
                  </div>
                  <div
                    aria-expanded={null}
                    className="collapse"
                  >
                    <div
                      className="card-body"
                    >
                      <div
                        className="flex-column nav card-header-undefined"
                        onKeyDown={[Function]}
                      >
                        <a
                          aria-current={null}
                          href="/console/settings/general"
                          onClick={[Function]}
                        >
                          General
                        </a>
                        <a
                          aria-current="page"
                          className="disabled active"
                          href="/"
                          onClick={[Function]}
                          style={Object {}}
                        >
                          Domain
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-md-9"
            >
              <div
                className="customization-form"
              >
                <div
                  className="loading"
                >
                  <i
                    className="fas fa-2x fa-sync-alt fa-spin"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
